src/main/java/com/squareup/javapoet/TypeSpec.java

```diff
@@ -225,24 +201,43 @@ void emit(CodeWriter codeWriter, String enumName) throws IOException {
     }
   }
 
-  private enum DeclarationType {
-    CLASS(ImmutableSet.<Modifier>of(), ImmutableSet.<Modifier>of()),
-    INTERFACE(ImmutableSet.of(Modifier.PUBLIC, Modifier.STATIC, Modifier.FINAL),
-        ImmutableSet.of(Modifier.PUBLIC, Modifier.ABSTRACT)),
-    ENUM(ImmutableSet.<Modifier>of(), ImmutableSet.<Modifier>of());
+  private enum Kind {
+    CLASS(
+        ImmutableSet.<Modifier>of(),
+        ImmutableSet.<Modifier>of(),
+        ImmutableSet.<Modifier>of(),
+        ImmutableSet.<Modifier>of()),
+
+    INTERFACE(
+        ImmutableSet.of(Modifier.PUBLIC, Modifier.STATIC, Modifier.FINAL),
+        ImmutableSet.of(Modifier.PUBLIC, Modifier.ABSTRACT),
```

Comment:
Looks like we'd need Java8-specific unit-tests (and use a Maven profile to skip them when running tests in a Java 7 VM)